tests/misc/check_init_priorities/CMakeLists.txt

```diff
@@ -8,10 +8,9 @@ set(output_file ${PROJECT_BINARY_DIR}/check_init_priorities_output.txt)
 add_custom_command(
 	COMMENT "Running check_init_priorities.py"
 	OUTPUT ${output_file}
-	DEPENDS ${BYPRODUCT_KERNEL_ELF_NAME}
-	COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/check_init_priorities.py
+	DEPENDS ${CHECK_INIT_PRIORITIES_DEPENDENCIES}
+	COMMAND ${CHECK_INIT_PRIORITIES_COMMAND}
```

Comment:
ok, I see the reason here for the global symbols.

However, placing such variables (including the genex expression) in the CMake cache simply to be able to re-use them in one specific CMake file testing the script is not ideal.

All other projects will now start to have this in their cache, just because we want to make a single test a bit cleaner.

If there is no general use of those vars besides this test, then I don't like to approve.

As I now understand the use-case a bit better, let me see if I can come up with a more clever approach.